### PR TITLE
media-libs/lib3mf: new package

### PR DESCRIPTION
**Pull request by waebbl**
Thanks to Scott Alfter (@salfter) who initially wrote the ebuild.

Reported-by: Scott Alfter <scott@alfter.us>
Closes: https://bugs.gentoo.org/686476
Package-Manager: Portage-2.3.96-r1, Repoman-2.3.22
Signed-off-by: Bernd Waibel <waebbl@gmail.com>

Replaces #12097 as noted in the comments there.

### CODE REVIEW

File: media-libs/lib3mf/lib3mf-1.8.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="Implementation of the 3D Manufacturing Format file standard"
+HOMEPAGE="https://3mf.io/"
+SRC_URI="https://github.com/3MFConsortium/$PN/archive/v$PV.tar.gz -> $P.tar.gz"
+KEYWORDS="~amd64 ~x86"
```
Comment (a17r) [inline media-libs/lib3mf/lib3mf-1.8.1.ebuild:11]: Per ebuild.skel I would expect `KEYWORDS` below `SLOT` (and in the same block).

File: media-libs/lib3mf/lib3mf-1.8.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+LICENSE="BSD"
+SLOT="0"
+
+RDEPEND="
+	dev-libs/libzip:=
+	sys-apps/util-linux
+	sys-libs/zlib:=
```
Comment (a17r) [inline media-libs/lib3mf/lib3mf-1.8.1.ebuild:19]: Introduction of a subslot on zlib was a mistake, it should not be used.

File: media-libs/lib3mf/lib3mf-1.8.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+"
+
+PATCHES=(
+	"${FILESDIR}/${P}-0001-Gentoo-specific-avoid-pre-stripping-library.patch"
+	"${FILESDIR}/${P}-0002-Add-library-link-dependencies.patch"
+	"${FILESDIR}/${P}-0003-Change-installation-include-dir.patch"
+	"${FILESDIR}/${P}-0004-Gentoo-specific-Remove-gtest-source-dir.patch"
+)
+
+IUSE="doc test"
```
Comment (a17r) [inline media-libs/lib3mf/lib3mf-1.8.1.ebuild:33]: Per ebuild.skel I would expect `IUSE` below `KEYWORDS` (and in the same block).

File: media-libs/lib3mf/lib3mf-1.8.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	)
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+
+	local DOCS=( CONTRIBUTING.md README.md )
+	use doc && DOCS+=( Lib3MF-1.pdf )
+	einstalldocs
```
Comment (a17r) [inline media-libs/lib3mf/lib3mf-1.8.1.ebuild:51]: Move these lines on top before `cmake_src_install` and you are able to drop the manual `einstalldocs`

File: media-libs/lib3mf/lib3mf-1.8.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="Implementation of the 3D Manufacturing Format file standard"
+HOMEPAGE="https://3mf.io/"
+SRC_URI="https://github.com/3MFConsortium/$PN/archive/v$PV.tar.gz -> $P.tar.gz"
+LICENSE="BSD"
+
+SLOT="0"
```
Comment (a17r) [inline media-libs/lib3mf/lib3mf-1.8.1.ebuild:13]: Per ebuild.skel, `LICENSE` is in the same block as `SLOT`... that didn't need change from the previous state. ;)

Comment (waebbl) [reply media-libs/lib3mf/lib3mf-1.8.1.ebuild:13]: Oh yes, you're right, there could be S inbetween SRC_URI and LICENSE. 
Although in my opinion, LICENSE is logically more tied to the block with SRC_URI. The block containing KEYWORDS and SLOT are metadata which is related to building... where (arch) and what (IUSE, SLOT, RESTRICT) to build. And many ebuilds have in in the upper block ;)
I'm gonna change it back.

**Pull request closed without merging**